[ONNX] Update repeat_interleave for dynamic repeats (#59979)

BowenBao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #63249 Enhance shape (#60335)
* #62765 [ONNX] Suppport torch.dot and torch.nn.utils.spectral_norm (#62596)
* **#62764 [ONNX] Update repeat_interleave for dynamic repeats (#59979)**
* #62763 [ONNX] Fix an issue that optimizations might adjust graph inputs unexpectedly. (#61280)
* #62762 [ONNX] Fix controlflow shape inference with contrib op (#60707)
* #62760 [ONNX] Fix for batchnorm training op mode (#52758)
* #62759 [ONNX] Remove aten parameter (#61652)
* #62758 [ONNX] Add support for opset14 in PT-ONNX exporter (#59486)
* #62757 [ONNX] Support lstm_cell symbolic (#61476)

Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

Differential Revision: [D30375179](https://our.internmc.facebook.com/intern/diff/D30375179)

shubhambhokare1 committed:
[ONNX] Update repeat_interleave for dynamic repeats (#59979)

Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

[ghstack-poisoned]

msaroufim commented:
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30375179).

BowenBao committed:
Update on "[ONNX] Update repeat_interleave for dynamic repeats (#59979)"


Fixes #58733

- Support dynamic interleave for cases with dynamic repeat values
- Moved repeat_interleave symbolic from opset 11 to opset 13, as sequence as output types for loop outputs is needed for this change

Co-authored-by: Shubham Bhokare <shubhambhokare@gmail.com>

Differential Revision: [D30375179](https://our.internmc.facebook.com/intern/diff/D30375179)

[ghstack-poisoned]
